dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java

```diff
@@ -157,10 +160,17 @@ public EventSearchParams map(EventOperationParams operationParams)
         .setProgram(program)
         .setProgramStage(programStage)
         .setOrgUnit(orgUnit)
+        .setAccessibleOrgUnits(
+            getUserAccessibleOrgUnits(
+                program,
+                user,
+                orgUnit,
+                operationParams.getOrgUnitSelectionMode(),
+                organisationUnitService::getOrganisationUnitWithChildren))
         .setTrackedEntity(trackedEntity)
         .setProgramStatus(operationParams.getProgramStatus())
         .setFollowUp(operationParams.getFollowUp())
-        .setOrgUnitSelectionMode(operationParams.getOrgUnitSelectionMode())
+        .setOrgUnitSelectionMode(getOrgUnitMode(orgUnit, operationParams.getOrgUnitSelectionMode()))
```

Comment:
If we extract this orgUnitMode to a variable and use it in the validateUser method we can remove the guard clauses there because we are sure that orgUnitMode is not null.

Doing this remove the duplication of logic that now is present in getOrgUnitMode method and validateUser method

Reply:
I rather not have the `validateUser` method return a list of org units, but I think you are right to extract this in a new variable, I'll create a new method for that.

Reply:
Sorry, that was confusing. I didn't mean validateUser method, I meant getUserAccessibleOrgUnits method